calicoctl/commands/datastore/migrate/migrateipam.go

```diff
@@ -120,6 +126,35 @@ func (m *migrateIPAM) PullFromDatastore() error {
 		if !ok {
 			return fmt.Errorf("Could not convert %+v to an AllocationBlock", item.Value)
 		}
+
+		// Update node names in the block to match the Kubernetes node
+		if m.nodeMap != nil {
+			for i, allocationAttribute := range block.Attributes {
+				// Update the node name if it has a corresponding Kubernetes node name
+				attrNodeName := strings.ReplaceAll(allocationAttribute.AttrSecondary["node"], ".", "-")
+				if nodeName, ok := m.nodeMap[attrNodeName]; ok {
+					block.Attributes[i].AttrSecondary["node"] = nodeName
+				}
+
+				// Update the handle ID for ipip tunnel addresses
+				if allocationAttribute.AttrPrimary != nil {
+					if strings.HasPrefix(*allocationAttribute.AttrPrimary, "ipip-tunnel-addr-") {
```

Comment:
What about the VXLAN and wireguard tunnel addresses? Probably need those here as well.